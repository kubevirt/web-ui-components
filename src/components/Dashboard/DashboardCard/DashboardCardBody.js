import React from 'react';
import PropTypes from 'prop-types';

import { CardBody } from 'patternfly-react';

import { Loading } from '../../Loading';

const DashboardCardBody = ({ LoadingComponent, isLoading, children, ...props }) => (
  <CardBody {...props}>{isLoading ? <LoadingComponent /> : children}</CardBody>
);

DashboardCardBody.defaultProps = {
  LoadingComponent: Loading,
  isLoading: false,
  children: null,
};

DashboardCardBody.propTypes = {
  LoadingComponent: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
  isLoading: PropTypes.bool,
  children: PropTypes.node,
};

export default DashboardCardBody;
