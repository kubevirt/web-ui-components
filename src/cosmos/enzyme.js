import createContext from 'react-cosmos-test/enzyme';

import paths from '../../config/paths';

export default function createTestContext(args) {
  args.proxies = args.proxies || []; // workaround of react-cosmos-test bug
  return createContext({
    ...args,
    cosmosConfigPath: `${paths.config}/cosmos.config.js`,
  });
}
