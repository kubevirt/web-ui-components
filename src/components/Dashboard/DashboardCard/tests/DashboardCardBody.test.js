import React from 'react';
import { shallow } from 'enzyme';

import DashboardCardBody from '../DashboardCardBody';

const testDashboardCardBody = (isLoading = false) => (
  <DashboardCardBody isLoading={isLoading}>content</DashboardCardBody>
);

describe('<DashboardCardBody />', () => {
  it('renders correctly', () => {
    const component = shallow(testDashboardCardBody());
    expect(component).toMatchSnapshot();
  });
  it('renders loading correctly', () => {
    const component = shallow(testDashboardCardBody(true));
    expect(component).toMatchSnapshot();
  });
});
