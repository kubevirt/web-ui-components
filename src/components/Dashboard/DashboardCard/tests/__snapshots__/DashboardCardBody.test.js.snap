// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardCardBody /> renders correctly 1`] = `
<CardBody
  className=""
>
  content
</CardBody>
`;

exports[`<DashboardCardBody /> renders loading correctly 1`] = `
<CardBody
  className=""
>
  <Loading
    text="Loading"
  />
</CardBody>
`;
